sql/src/main/java/io/crate/execution/dml/ShardRequestExecutor.java

```diff
@@ -98,16 +101,21 @@ public ShardRequestExecutor(ClusterService clusterService,
     }
 
     public void execute(RowConsumer consumer, Row parameters, SubQueryResults subQueryResults) {
```

Comment:
Would it make sense to split this class into 2 separate executors one for counting, one to retrieve return values ? Or maybe expose the possibility to handle ActionListener to the caller and leave the way the return values are handled to the outside of this class ?

Reply:
Another option would be to leave conversion helper methods here (they can be all static afaik) and built the `MultiActionListener` at the caller (or using static methods which may stay here). I think most of the stuff here is already very generic so splitting this up into 2 implementations just for wrangling the results sounds a bit too much to me.

Reply:
Would the current state be sufficient ?